### PR TITLE
Add csv output also for singe phase flow and unify a bit between different statistics outputs

### PR DESCRIPTION
**Pull request by paveltomin**
Add csv output also for singe phase flow and unify a bit between different statistics outputs

Example of csv output:
```
Time [s],Min pressure [Pa],Average pressure [Pa],Max pressure [Pa],Min delta pressure [Pa],Max delta pressure [Pa],Min temperature [Pa],Average temperature [Pa],Max temperature [Pa],Total dynamic pore volume [rm^3],Total fluid mass [kg]
1e+06,1.01533e+07,1.12267e+07,1.23004e+07,0,0,300,300,300,1.5625e+06,1.56334e+09
2e+06,1.46268e+07,1.99449e+07,2.90815e+07,0,0,300,300,300,1.59139e+06,1.59837e+09
3e+06,1.65824e+07,2.39635e+07,3.36875e+07,2.94006e+06,1.8315e+07,300,300,300,1.60503e+06,1.61491e+09
4e+06,1.72833e+07,2.56001e+07,3.50638e+07,4.8957e+06,2.2921e+07,300,300,300,1.61063e+06,1.6217e+09
5e+06,1.75221e+07,2.62623e+07,3.55192e+07,5.59656e+06,2.42972e+07,300,300,300,1.61292e+06,1.62447e+09
6e+06,1.76052e+07,2.65339e+07,3.56796e+07,5.83534e+06,2.47527e+07,300,300,300,1.61386e+06,1.62562e+09
7e+06,1.76353e+07,2.6647e+07,3.57389e+07,5.91847e+06,2.49131e+07,300,300,300,1.61426e+06,1.6261e+09
8e+06,1.76466e+07,2.66947e+07,3.57617e+07,5.94857e+06,2.49723e+07,300,300,300,1.61443e+06,1.6263e+09
9e+06,1.76511e+07,2.6715e+07,3.57708e+07,5.95992e+06,2.49952e+07,300,300,300,1.6145e+06,1.62639e+09
1e+07,1.76529e+07,2.67237e+07,3.57745e+07,5.96435e+06,2.50042e+07,300,300,300,1.61453e+06,1.62642e+09
```

integratedTests: https://github.com/GEOS-DEV/integratedTests/pull/83

### CODE REVIEW

File: src/coreComponents/physicsSolvers/FieldStatisticsBase.hpp [hunk 1]
```diff
@@ -85,11 +87,25 @@ class FieldStatisticsBase : public TaskBase
                              getDataContext(),
                              m_solverName, LvArray::system::demangleType< SOLVER >() ),
                    InputError );
+
+    // create dir for output
+    if( getLogLevel() > 0 )
```
Comment (MelReyCG) [inline src/coreComponents/physicsSolvers/FieldStatisticsBase.hpp]: I think that we should not couple the CSV and the log writing here.
I suggest to add a new attribute for this purpose (like `writeCSV="1"`).

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp [hunk 2]
```diff
@@ -59,12 +59,23 @@ void SinglePhaseStatistics::registerDataOnMesh( Group & meshBodies )
       region.registerWrapper< RegionStatistics >( viewKeyStruct::regionStatisticsString() ).
         setRestartFlags( RestartFlags::NO_WRITE );
       region.excludeWrappersFromPacking( { viewKeyStruct::regionStatisticsString() } );
+
+      // write output header
+      if( getLogLevel() > 0 && MpiWrapper::commRank() == 0 )
```
Comment (MelReyCG) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp]: Here

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp [hunk 3]
```diff
@@ -230,23 +243,29 @@ void SinglePhaseStatistics::computeRegionStatistics( MeshLevel & mesh,
...
+    GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}, {} (time {} s): Delta pressure (min, max): {}, {} Pa",
+                                        getName(), regionNames[i], time, regionStatistics.minDeltaPressure, regionStatistics.maxDeltaPressure ) );
+    GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}, {} (time {} s): Temperature (min, average, max): {}, {}, {} K",
+                                        getName(), regionNames[i], time, regionStatistics.minTemperature, regionStatistics.averageTemperature, regionStatistics.maxTemperature ) );
+    GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}, {} (time {} s): Total dynamic pore volume: {} rm^3",
+                                        getName(), regionNames[i], time, regionStatistics.totalPoreVolume ) );
+    GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}, {} (time {} s): Total fluid mass: {} kg",
+                                        getName(), regionNames[i], time, regionStatistics.totalMass ) );
+
+    if( getLogLevel() > 0 && MpiWrapper::commRank() == 0 )
```
Comment (MelReyCG) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseStatistics.cpp]: And here

**Review by MelReyCG — CHANGES_REQUESTED** [review]
Looks good to me for the CSV writing part, but I think that we should not couple the CSV and the log writing.
I suggest to add a new attribute for this purpose (like `writeCSV="1"`).

File: src/coreComponents/physicsSolvers/FieldStatisticsBase.hpp (hunk 1 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/FieldStatisticsBase.hpp]: ok done, the reason it was done with logLevel is because I wanted to avoid rebase...

**Review by MelReyCG — APPROVED** [review]
I think that is cleaner this way. Your PR looks good to me :)

**Comment by MelReyCG** [comment]
Be careful, there are a few `.rst` that need to be updated to take this new attribute into account:
```
- src/coreComponents/schema/docs/CompositionalMultiphaseStatistics.rst
- src/coreComponents/schema/docs/CompositionalMultiphaseWell.rst
- src/coreComponents/schema/docs/SinglePhaseStatistics.rst
- src/coreComponents/schema/docs/SinglePhaseWell.rst
- src/coreComponents/schema/docs/SolidMechanicsStatistics.rst
```
I can push them in your branch if you want.

**Comment by paveltomin** [comment]
> Be careful, there are a few `.rst` that need to be updated to take this new attribute into account:
> 
> ```
> - src/coreComponents/schema/docs/CompositionalMultiphaseStatistics.rst
> - src/coreComponents/schema/docs/CompositionalMultiphaseWell.rst
> - src/coreComponents/schema/docs/SinglePhaseStatistics.rst
> - src/coreComponents/schema/docs/SinglePhaseWell.rst
> - src/coreComponents/schema/docs/SolidMechanicsStatistics.rst
> ```
> 
> I can push them in your branch if you want.

yes please push, thanks

**Comment by MelReyCG** [comment]
Done!
Don't forget to `make install` to generate these ones (with `ENABLE_DOCS`, `ENABLE_XML_UPDATES`, and `ENABLE_SPHINX` on)

**Comment by paveltomin** [comment]
this PR should go after https://github.com/GEOS-DEV/GEOS/pull/2904

**Pull request merged**